node/src/accountant/scanners/pending_payable_scanner/tx_receipt_interpreter.rs

```diff
@@ -0,0 +1,732 @@
+                scan_report.register_new_failure(failed_tx);
+            }
+            TxByTable::FailedPayable(failed_tx) => {
+                if failed_tx.reason != FailureReason::PendingTooLong {
+                    unreachable!(
+                        "Transaction is both pending and failed (failure reason: '{:?}'). Should be \
+                        possible only with the reason 'PendingTooLong'",
+                        failed_tx.reason
+                    )
+                }
+                let replacement_tx = sent_payable_dao
+                    .retrieve_txs(Some(RetrieveCondition::ByNonce(vec![failed_tx.nonce])));
+                let replacement_tx_hash = replacement_tx
+                    .get(0)
+                    .unwrap_or_else(|| {
+                        panic!(
+                            "Attempted to display a replacement tx for {:?} but couldn't find \
+                        one in the database",
+                            failed_tx.hash
+                        )
+                    })
+                    .hash;
+                warning!(
+                    logger,
+                    "Failed tx {:?} on a recheck was found pending on its receipt unexpectedly. \
+                    It was supposed to be replaced by {:?}",
+                    failed_tx.hash,
+                    replacement_tx_hash
+                );
+                scan_report.register_rpc_failure(FailedValidationByTable::FailedPayable(
+                    FailedValidation::new(
+                        failed_tx.hash,
+                        BlockchainErrorKind::Internal(InternalErrorKind::PendingTooLongNotReplaced),
+                        failed_tx.status,
+                    ),
+                ))
+            }
+        }
+        scan_report
+    }
+
+    fn elapsed_in_ms(timestamp: SystemTime) -> u128 {
+        timestamp
+            .elapsed()
+            .expect("time calculation for elapsed failed")
+            .as_millis()
+    }
+
+    fn handle_tx_confirmation(
+        mut scan_report: ReceiptScanReport,
+        tx: TxByTable,
+        tx_block: TxBlock,
+        logger: &Logger,
+    ) -> ReceiptScanReport {
+        match tx {
+            TxByTable::SentPayable(sent_tx) => {
+                info!(
+                    logger,
+                    "Pending tx {:?} was confirmed on-chain", sent_tx.hash,
+                );
+
+                let completed_sent_tx = SentTx {
+                    status: TxStatus::Confirmed {
+                        block_hash: format!("{:?}", tx_block.block_hash),
+                        block_number: tx_block.block_number.as_u64(),
+                        detection: Detection::Normal,
+                    },
+                    ..sent_tx
+                };
+                let tx_confirmation = NormalTxConfirmation {
+                    tx: completed_sent_tx,
+                };
+                scan_report.register_confirmed_tx(tx_confirmation);
```

Comment:
I would greatly appreciate it if we could simplify the implementation by having a single function named `register_confirmed_tx(tx, confirmation_type)`. This approach would eliminate the need for using structs like `NormalTxConfirmation` and `TxReclaim`. Instead, we would have an enum which will tell this part of the story.